packages/downloader/download.py

```
@@ -0,0 +1,126 @@
+    
+    try:
+        # Don't write finished status from progress_hook - let postprocessor_hook handle it
+        if data.get("status") == "finished":
+            return
+        
+        # Write to temporary file first
+        with open(temp_path, 'w') as wo:
+            json.dump(data, wo)
+        # Atomically move to final location
+        os.rename(temp_path, final_path)
+    except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)